### PR TITLE
Prevector: fix some compile errors. Make resize faster

### PR DESCRIPTION
**Pull request by sinetek**
* speed up Unserialize_impl for prevector (Akio Nakamura). BTC #12324

* Gets rid of many compile warnings. (see next point)

* Lift prevector default vals to the member declaration (Ben Woosley)

### CODE REVIEW

File: src/serialize.h [hunk 1]
```diff
@@ -768,7 +768,7 @@ void Unserialize_impl(Stream& is, prevector<N, T>& v, const V&)
         nMid += 5000000 / sizeof(T);
         if (nMid > nSize)
             nMid = nSize;
-        v.resize(nMid);
+        v.resize_uninitialized(nMid);
         for (; i < nMid; i++)
```
Comment (codeofalltrades) [inline src/serialize.h]: The btc issue has it ++i
`for (; i < nMid; ++i)`

Comment (sinetek) [reply src/serialize.h]: both were okay. done anyway.

**Comment by codeofalltrades** [comment]
TravisCI keeps failing with the same error.
Build - x86_64 Linux [GOAL: install] [trusty] [no functional tests, no depends, only system libs]

```Making install in src
make[1]: Entering directory `/home/travis/build/Veil-Project/veil/build/veil-x86_64-unknown-linux-gnu/src'
make[2]: Entering directory `/home/travis/build/Veil-Project/veil/build/veil-x86_64-unknown-linux-gnu/src'
  CXX      veild-veild.o
  CXX      libbitcoin_server_a-addrdb.o
  CXX      libbitcoin_server_a-addrman.o
  CXX      libbitcoin_server_a-bloom.o
  CXX      libbitcoin_server_a-blockencodings.o
  CXX      libbitcoin_server_a-chain.o
  CXX      libbitcoin_server_a-checkpoints.o
In file included from ./libzerocoin/Params.h:17:0,
                 from ./libzerocoin/Coin.h:17,
                 from ./libzerocoin/Accumulator.h:16,
                 from ./veil/zerocoin/accumulatormap.h:8,
                 from ./chain.h:9,
                 from chain.cpp:6:
./libzerocoin/bignum.h: In static member function ‘static CBigNum CBigNum::randBignum(const CBigNum&)’:

./libzerocoin/bignum.h:793:58: error: ‘CHAR_BIT’ was not declared in this scope
         size_t size = (mpz_sizeinbase (range.bn, 2) + CHAR_BIT-1) / CHAR_BIT;
                                                          ^
./libzerocoin/bignum.h: In member function ‘std::vector<unsigned char> CBigNum::getvch() const’:
./libzerocoin/bignum.h:902:52: error: ‘CHAR_BIT’ was not declared in this scope
         size_t size = (mpz_sizeinbase (bn, 2) + CHAR_BIT-1) / CHAR_BIT;
                                                    ^```

File: src/prevector.h [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Copyright (c) 2015-2019 The Bitcoin Core developers
+// Copyright (c) 2015-2018 The Bitcoin Core developers
```
Comment (CaveSpectre11) [inline src/prevector.h]: Unless it's a straight cherry pick from another project (showing their dev commit], add the veil copyright to any module touched.

Comment (sinetek) [reply src/prevector.h]: yea its straight from bitcoin, so ... does it really matter?

**Pull request merged**